### ISSUE TITLE
[MASSEMBLY-597] Duplicate files added to archive when present in both dependencyset and fileset of the same assembly

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-assembly-plugin#754]
**[Eric Daigneault](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=edaigno)** opened **[MASSEMBLY-597](https://issues.apache.org/jira/browse/MASSEMBLY-597?redirect=false)** and commented

Comment added to parent bug, could not re-open the issue so I created this sub-task instead.  Comment copied here for convenience :

---

Either version 2.2.2 of the Assembly plugin has a regression or the issue has not been fixed.  I am currently creating zip files from the assembly that contains duplicate jar files.

Some background :  I am using the assembly to create a zip file of an application.  Why Zip ? because the application requires multiple configuration files and spans larger than just Java so I cannot use a uber Jar (I would prefer that but technical limitations out of my control force me otherwise).

This package is built correctly.

To make life simpler I made it possible to extend this package with a project that can add extra classes (plugins and such) and configuration from a standard layout in the project that gets picked up by the assembly and merged with the previous vanilla package.  the end result is a fully assembled zip file with all the customized parts in the right place.

Now, behaviour changed across version wheras previously I would overwrite the original package with the content of the new one, now I have to start from the overrides and complete withe the original files.  Easily fixed through changing the order I declared the filesets in the assembly.  When using filesets strictly nothing gets added twice, though the replace was changed to skip which is a bit counter intuitive.

BUT....

Since I extend some parts of the original system, I will share some dependencies with this one.  These dependencies are already present in the original package and get added through a file set (I get the package and unzip it in a temporary folder where I integrate it`s content inthe assembly through a fileset).

I also add the dependencies of the new customized project, and these get added twice.

I think there is a regression here when filesets and dependencysets interact where if files are present in both they get added twice in the zip file.  Now I would re-open this current task but it seems I cannot so I will open a sub task instead.

---

**Affects:** 2.2.2

**Attachments:**
- [ma597.zip](https://issues.apache.org/jira/secure/attachment/12714361/ma597.zip) (_3.85 kB_)

**Issue Links:**
- [MASSEMBLY-285](https://issues.apache.org/jira/browse/MASSEMBLY-285) regression: duplicate files added to the assembly

- [MASSEMBLY-726](https://issues.apache.org/jira/browse/MASSEMBLY-726) Fix artifact inclusion/exclusion filtering

2 votes, 5 watchers

**Comment by jira-importer** [comment]
**[Dennis Lundberg](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=dennisl@apache.org)** commented

Can you please provide a small sample project that can be used to verify this issue?

**Comment by jira-importer** [comment]
**[Michal Michalski](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=kontomen)** commented

I have the same issue with 2.3 version.

**Comment by jira-importer** [comment]
**[Dennis Lundberg](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=dennisl@apache.org)** commented

Michal,

Can you provide a sample project for this issue?

**Comment by jira-importer** [comment]
**[Dzmitry Paulenka](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=fantast)** commented

Sample projects, demonstrating the issue.

**Comment by jira-importer** [comment]
**[Dzmitry Paulenka](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=fantast)** commented

I'm attaching sample projects, demonstrating the issue. First you need to mvn install "dependency". Than do mvn install on "project". Now you can see "target/project-full.jar", produced by assembly-plugin. So the issue is that it contains duplicates files under "/subdir" directory. BTW, you can see, that it didn't duplicate file "/file.txt" in the root of the jar, and used the correct one from project resources, which is good, but strange.

**Comment by jira-importer** [comment]
**[Kristian Rosenvold](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=krosenvold)** commented

Using the enclosed project it looks correct to me. I tried both maven 2.2.1 and 3.x and the output looks entirely correct. This was on mac/jdk8. I even tried a fair number of different versions of the assembly plugin, and they all look correct. Exactly what are you seeing ?

Actually, you might want to try assembly version 2.5 (or 2.5.1), since there was a windows-related path bug that was fixed for 2.5

Addendum: This walks like PLXCOMP-254, talks like PLXCOMP-254 and probably is PLXCOMP-254
